pkg/kube/client.go

```diff
@@ -1206,3 +1212,37 @@ func SetRevisionForTest(c CLIClient, rev string) CLIClient {
 	tc.revision = rev
 	return tc
 }
+
+func findIstiodMonitoringPort(pod *v1.Pod) int {
+	if v, ok := pod.GetAnnotations()["prometheus.io/port"]; ok {
+		if port, err := strconv.Atoi(v); err == nil {
+			return port
+		}
+	}
+	for _, container := range pod.Spec.Containers {
+		if container.Name != discoveryContainerName {
+			continue
+		}
+		argsStr := strings.Join(container.Args, " ")
+		if !strings.Contains(argsStr, "monitoringAddr") {
+			continue
+		}
+		args := strings.Split(argsStr, " ")
+		for i, arg := range args {
+			var port string
+			if strings.HasPrefix(arg, "--monitoringAddr=") {
+				addr := strings.TrimSpace(strings.TrimPrefix(arg, "--monitoringAddr="))
+				_, port, _ = net.SplitHostPort(addr)
+			} else if arg == "--monitoringAddr" && i+1 < len(args) {
+				addr := strings.TrimSpace(args[i+1])
+				_, port, _ = net.SplitHostPort(addr)
+			}
+			if len(port) > 0 {
+				if p, err := strconv.Atoi(port); err == nil {
+					return p
+				}
+			}
+		}
+	}
```

Comment:
It doesn't seem necessary here, if someone change `monitoringAddr`, annotation must be changed the same time.

Reply:
This is to avoid users who do not have Prometheus installed in their clusters, and no prom annotation is set when they have `enablePrometheusMerge=false`.